Formula/helmfile.rb

```diff
@@ -0,0 +1,22 @@
+class Helmfile < Formula
+  desc "Deploy Kubernetes Helm Charts"
+  homepage "https://github.com/roboll/helmfile"
+  url "https://github.com/roboll/helmfile/archive/v0.18.0.tar.gz"
+  sha256 "6fc6ab175a746f895b33d13f169293f55425bdc4efad33a25cf226b14b2e6f6f"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["CGO_ENABLED"] = "0"
+    ENV["GOPATH"] = buildpath
+    (buildpath/"src/github.com/roboll/helmfile").install buildpath.children
+    cd "src/github.com/roboll/helmfile" do
+      system "go", "build", "-ldflags", "-X main.Version=v#{version}", "-v", "github.com/roboll/helmfile"
+      bin.install "helmfile"
+    end
+  end
+
+  test do
+    assert_match /v#{version}$/, shell_output("#{bin}/helmfile --version")
```

Comment:
We need a test that exercises the some of the functionality of the app.

Reply:
Helmfile is a coordinating wrapper around Helm, so I've added helm as a dependency, and in the tests initialized it and used it. I'm unclear on best practices for declaring this sort of dependency. Since it's runtime-only, but may be installed outside of Homebrew, is it still appropriate to list as a dependency? Or should it just be a test dependency?

My update to this PR treats it as a normal dependency, but please let me know if this needs more adjusting.

Reply:
`depends_on "kubernetes-helm" => :recommended` seems like the right way to go

Reply:
It looks like the best practices for new formulae just changed 9 days ago (Homebrew/brew#4170) and there are now errors for both `:recommended` and `:optional` dependencies:

```
→ brew audit --new-formula helmfile
helmfile:
  * Formulae should not have [:recommended] dependencies.
Error: 1 problem in 1 formula
```

```
→ brew audit --new-formula helmfile
helmfile:
  * Formulae should not have [:optional] dependencies.
Error: 1 problem in 1 formula
```

Should this just be merged as-is? Should a follow-up PR add the option?